### PR TITLE
Update azure sdk

### PR DESCRIPTION
**Pull request by guilhem**
### Description of your changes

Upgrade Azure SDK without many changes.
This should permit future evolves like private endpoint and private DNS group.

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/clients/network/network.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ package network
 import (
 	"reflect"
 
-	networkmgmt "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	networkmgmt "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-07-01/network"
```
Comment (muvaf) [inline pkg/clients/network/network.go:22]: @guilhem do you know if there are any behavior changes between the versions?

Comment (guilhem) [reply pkg/clients/network/network.go:22]: @muvaf I didn't see that much change. Only new resources exposed :)
But to be honest, I can remove this change and bump version only when needed in another PR.

Comment (muvaf) [reply pkg/clients/network/network.go:22]: Got it. Do you plan on making that new change soon?

Comment (guilhem) [reply pkg/clients/network/network.go]: @muvaf done :)

**Comment by ulucinar** [comment]
Hi @guilhem,
I'm going over the open PRs/issues and I would like to thank you for your work here. Are you still interested in working on this PR?

**Comment by guilhem** [comment]
@ulucinar you can take this work :)

**Comment by ulucinar** [comment]
Thank you @guilhem. We are closing this PR via #326 which also follows your approach here.

**Pull request closed without merging**